src/transformers/models/whisper/generation_whisper.py

```diff
@@ -733,12 +707,35 @@ def generate(
             if (prompt_ids is not None and generation_config.prompt_condition_type == "first-segment")
             else current_segments
         )
-        sequences = _pad_to_max_length(final_segments, generation_config.pad_token_id, padding="right")
+        sequences = _pad_to_max_length(final_segments, pad_token_id = generation_config.pad_token_id, padding="right")
 
         # 8. If we return all segments, the predicted output sequences are put under `"sequences"`.
         if return_segments:
             return {"sequences": sequences, "segments": final_segments}
 
+        if is_shortform: 
+            sequences = torch.cat([decoder_input_ids.to(sequences.device), sequences], dim=-1)
```

Comment:
Don't know why we still do this rather than not stripping the input ids in the first place https://github.com/huggingface/transformers/pull/30984#discussion_r1618595148